src/Simulation/TargetDefinitions/Decompositions/SWAPFromSinglyControlled.qs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Intrinsic {
+    open Microsoft.Quantum.Diagnostics;
+
+    /// # Summary
+    /// Applies the SWAP gate to a pair of qubits.
+    ///
+    /// # Description
+    /// \begin{align}
+    ///     \operatorname{SWAP} \mathrel{:=}
+    ///     \begin{bmatrix}
+    ///         1 & 0 & 0 & 0 \\\\
+    ///         0 & 0 & 1 & 0 \\\\
+    ///         0 & 1 & 0 & 0 \\\\
+    ///         0 & 0 & 0 & 1
+    ///     \end{bmatrix},
+    /// \end{align}
+    ///
+    /// where rows and columns are ordered as in the quantum concepts guide.
+    ///
+    /// # Input
+    /// ## qubit1
+    /// First qubit to be swapped.
+    /// ## qubit2
+    /// Second qubit to be swapped.
+    ///
+    /// # Remarks
+    /// Equivalent to:
+    /// ```qsharp
+    /// CNOT(qubit1, qubit2);
+    /// CNOT(qubit2, qubit1);
+    /// CNOT(qubit1, qubit2);
+    /// ```
+    @EnableTestingViaName("Test.TargetDefinitions.SWAP")
+    operation SWAP (qubit1 : Qubit, qubit2 : Qubit) : Unit is Adj + Ctl {
+        body (...) {
+            ApplyUncontrolledSWAP(qubit1, qubit2);
+        }
+        adjoint self;
+        controlled (ctls, ...) {
+            if (Length(ctls) == 0) {
+                ApplyUncontrolledSWAP(qubit1, qubit2);
+            }
+            else {
+                within {
```

Comment:
Nit: is it worth using `within` here? I'd expect folks to be used to decomposing SWAP into three 'independent' CNOT operations rather than thinking about the outer CNOTs as body/adjoint pair.

Reply:
I tend to agree, and I originally had the three independent `CNOT` version both here and in the normal decomposition of `SWAP` used by QSharp.Core, but ran into a merge conflict with https://github.com/microsoft/qsharp-runtime/pull/458 done by @msoeken. Given that this change was merged before mine, I didn't want to undo it, so I used that formulation for both versions of the `SWAP` decomposition.

Reply:
but now looking again at @msoeken 's change, I think I might be able to make this decomposition more efficient and still correct if I remove the controls from `CNOT` in the within. @msoeken is that right?

Reply:
I confirmed with @bettinaheim that the controls for the conjugation form are not needed, so went ahead and removed them.